test/pullrequest/e2e_test.go

```diff
@@ -0,0 +1,170 @@
+package pullrequest
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/jenkins-x/go-scm/scm"
+)
+
+var (
+	proxy = flag.Bool("use-fake-scm-proxy", true, "whether to use fake in-memory proxy instead of using real SCM servers")
+)
+
+func TestPullRequest(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", t.Name())
+	if err != nil {
+		log.Fatalf("ioutil.TempDir: %v", err)
+	}
+	t.Logf("Test dir: %s", tmpdir)
+	defer os.RemoveAll(tmpdir)
+
+	// Make sure command is installed locally to resolve any dependencies.
+	// We need to do this first since the run might try to pull in new
+	// dependencies through the proxy.
+	install := exec.Command("go", "get", "-u", "github.com/tektoncd/pipeline/cmd/pullrequest-init")
+	t.Log(install)
+	if out, err := install.CombinedOutput(); err != nil {
+		t.Fatal(string(out))
+	}
+
+	for _, tc := range []struct {
+		name string
+		url  string
+
+		// Location of fake SCM data to use for proxy.
+		scmData string
+
+		// Auth token to use.
+		token string
+		// If auth token present, fmt template for Authorization header to check.
+		authHeader string
+	}{
+		{
+			name: "github-default",
+			// Must be http so we can intercept the request in the proxy.
+			url:        "http://github.com/wlynch/test/pulls/1",
+			scmData:    filepath.Join("testdata", "scm", "api.github.com"),
+			token:      envToken("GITHUB_TOKEN"),
+			authHeader: "Bearer %s",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			dir := filepath.Join(tmpdir, tc.name)
+			if err := os.Mkdir(dir, 0755); err != nil {
+				t.Fatalf("os.Mkdir: %v", err)
+			}
+
+			if tc.token != "" {
+				os.Setenv("AUTH_TOKEN", tc.token)
+				defer os.Unsetenv("AUTH_TOKEN")
```

Comment:
_ooooooooohhhhhhhh_ you're running the pull request init binary directly!!! Everything makes a lot more sense now. (What did you think it was doing beforehand you might ask? I guess I assumed you were running an entire Task or something but clearly that isn't happening XD)

I think there are 2 things that could be done to help an easily confused skimmer like me:
1. Mention in that readme that you are running the pull request binary directly (i saw that there was an http server running and that was the only potential other "binary" i could wrap my head around for a while)
2. maybe instead of `runCommand` you could call the function something like `runPRBinary` or something? (even if runCommand is technically correct)?

Reply:
Done.